src/test/java/org/prebid/server/bidder/videobyte/VideobyteBidderTest.java

```diff
@@ -0,0 +1,286 @@
+
+        // then
+        assertThat(result.getValue()).hasSize(2)
+                .extracting(HttpRequest::getPayload)
+                .extracting(BidRequest::getImp)
+                .extracting(List::size)
+                .containsOnly(1);
+        assertThat(result.getErrors()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldCreateRequestsWithCorrectUri() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(givenImp(ext -> ext.publisherId("1 23"), identity()),
+                givenImp(ext -> ext.publisherId("456").placementId("a/bc"), identity()),
+                givenImp(ext -> ext.publisherId("789").placementId("dce").networkId("A?a=BC"), identity()));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = bidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getValue()).hasSize(3)
+                .extracting(HttpRequest::getUri)
+                .containsExactly(ENDPOINT_URL + "?source=pbs&pid=" + HttpUtil.encodeUrl("1 23"),
+                        ENDPOINT_URL + "?source=pbs&pid=456&placementId=" + HttpUtil.encodeUrl("a/bc"),
+                        ENDPOINT_URL + "?source=pbs&pid=789&placementId=dce&nid=" + HttpUtil.encodeUrl("A?a=BC"));
+        assertThat(result.getErrors()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldCreateRequestsWithOriginHeaderIfSiteDomainIsPresent() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(Site.builder().domain("domain").build(),
+                givenImp(identity()), givenImp(identity()));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = bidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getValue()).hasSize(2)
+                .extracting(HttpRequest::getHeaders)
+                .flatExtracting(MultiMap::entries)
+                .hasSize(6)
+                .elements(2, 5)
+                .extracting(Map.Entry::getKey, Map.Entry::getValue)
+                .containsOnly(tuple(HttpUtil.ORIGIN_HEADER.toString(), "domain"));
+        assertThat(result.getErrors()).isEmpty();
+    }
+
+    @Test
+    public void makeHttpRequestsShouldCreateRequestsWithRefererHeaderIfSiteRefIsPresent() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(Site.builder().ref("referer").build(),
+                givenImp(identity()), givenImp(identity()));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = bidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getValue()).hasSize(2)
+                .extracting(HttpRequest::getHeaders)
+                .flatExtracting(MultiMap::entries)
+                .hasSize(6)
+                .elements(2, 5)
+                .extracting(Map.Entry::getKey, Map.Entry::getValue)
+                .containsOnly(tuple(HttpUtil.REFERER_HEADER.toString(), "referer"));
```

Comment:
Pls, replace all calls to `containsOnly` with calls to `containsExactly`.

Reply:
Do you really wanna replace
```
.containsOnly(tuple(HttpUtil.REFERER_HEADER.toString(), "referer"))
```
with
```
.containsExactly(tuple(HttpUtil.REFERER_HEADER.toString(), "referer"),
                        tuple(HttpUtil.REFERER_HEADER.toString(), "referer"))
```

Reply:
1) You should not rely on element's indices, it makes tests too fragile.
2) Yes, thats more explicit.

Reply:
Pls, check new realisation. If it ok, resolve conversation.